pkg/operator/controller/ingress/dns.go

```diff
@@ -34,7 +34,7 @@ func (r *reconciler) ensureWildcardDNSRecord(ic *operatorv1.IngressController, p
 		return false, nil, nil
 	}
 
-	if !manageDNSForDomain(ic.Status.Domain, platformStatus, dnsConfig) {
+	if !manageDNSForDomain(ic.Status.Domain, platformStatus, dnsConfig) && ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy != operatorv1.UnmanagedLoadBalancerDNS {
```

Comment:
Shouldn't this be an or? If the LB's DNS policy is managed but the domain is unmanaged, I'd expect the operator not to create a wildcard record (and vice versa)

Reply:
So there is a variation in the implementation of this and #761. The new added field controls DNS record management on the said cloud provider only i.e. the `DNSRecord` CR is still created in an `Unmanaged` state. The CR is created so that user/admin can use the information (`target`) present in it to create the required DNS records manually. ([REF](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/configurable-dns-loadbalancerservice.md#implementation-detailsnotesconstraints))

In PR #761, the logic that was implemented was that in addition to skipping creation of DNS records on the cloud provider it also skipped the creation of the `DNSRecord` CR. 

It was decided we could converge the 2 implementations and settle with the current solution, where the `DNSRecord` CR is created in an `Unmanaged` state. I'll be raising the needed PRs post getting this merged.

Reply:
> It was decided we could converge the 2 implementations and settle with the current solution, where the DNSRecord CR is created in an Unmanaged state. I'll be raising the needed PRs once this is merged.

So, the missing peace here is just setting `DNSRecord` as `Unmanaged` if the domain doesn't match and IC is `Managed`, right? Because if IC is `Unmanaged` we will create `Unmanaged` `DNSRecord` in any case (matching the domain or not).
Isn't it like half way implemented? I mean if a followup PR is expected, why not leave this condition untouched - letting IC not create `DNSRecord` if the domain doesn't match. Or just add the missing peace I said about above.

Reply:
Hmm, I just checked the change seems small as well so might as well add it. 
cc: @gcs278